app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEActivity.kt

```diff
@@ -0,0 +1,789 @@
+        if (sessionManagerServiceImpl.isUserLoggedIn()) {
+            presenter.setOpeningFilterToNone()
+            if (initSearchNeeded) {
+                presenter.onDestroy()
+            }
+        }
+        super.onPause()
+    }
+
+    override fun onDestroy() {
+        if (sessionManagerServiceImpl.isUserLoggedIn()) {
+            presenter.onDestroy()
+            FilterManager.getInstance().clearEnrollmentStatus()
+            FilterManager.getInstance().clearEventStatus()
+            FilterManager.getInstance().clearEnrollmentDate()
+            FilterManager.getInstance().clearWorkingList(true)
+            FilterManager.getInstance().clearSorting()
+            FilterManager.getInstance().clearAssignToMe()
+            FilterManager.getInstance().clearFollowUp()
+            presenter.clearOtherFiltersIfWebAppIsConfig()
+        }
+        super.onDestroy()
+    }
+
+    override fun onBackPressed() {
+        viewModel.onBackPressed(
+            isPortrait(),
+            viewModel.searchOrFilterIsOpen(),
+            this.isKeyboardOpened(),
+            {
+                super.onBackPressed()
+            },
+            {
+                if (viewModel.filterIsOpen()) {
+                    showHideFilterGeneral()
+                }
+                viewModel.setPreviousScreen()
+            },
+            {
+                hideKeyboard()
+            },
+        )
+    }
+
+    override fun onBackClicked() {
+        onBackPressed()
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putSerializable(Constants.QUERY_DATA, viewModel.queryData as Serializable)
+        outState.putString(CURRENT_SCREEN, currentContent!!.name)
+    }
+
+    private fun openSyncDialog() {
+        val contextView = findViewById<View>(R.id.navigationBar)
+        SyncStatusDialog.Builder()
+            .withContext(this, null)
+            .withSyncContext(
+                SyncContext.TrackerProgram(initialProgram!!),
+            )
+            .onDismissListener(object : OnDismissListener {
+                override fun onDismiss(hasChanged: Boolean) {
+                    if (hasChanged) viewModel.refreshData()
+                }
+            })
+            .onNoConnectionListener {
+                Snackbar.make(
+                    contextView,
+                    R.string.sync_offline_check_connection,
+                    Snackbar.LENGTH_SHORT,
+                ).show()
+            }
+            .show("PROGRAM_SYNC")
+    }
+
+    override fun updateFilters(totalFilters: Int) {
+        binding.totalFilters = totalFilters
+        binding.executePendingBindings()
+        viewModel.updateActiveFilters(totalFilters > 0)
+        viewModel.refreshData()
+    }
+
+    private fun initSearchParameters() {
+        initSearchScreen(
+            binding.searchContainer,
+            viewModel,
+            initialProgram,
+            tEType!!,
```

Comment:
Couldn't this lead to a crash if there is no teType?

Reply:
It doesn't look like it's going to be a null value, the reason it's a nullable is because the platform `intent.getStringExtra` is marked as nullable. But in our scenario we always pass it when creating this activity.

To handle this, I have converted the variable to a `lateinit` and set the default intent extra in case it's null to empty string. At least we don't have to force cast to non-null now.

Reply:
I have also cleaned up some of the other non-null assertions